### ISSUE TITLE
prefix option install modules differently and not in the right way

### ISSUE THREAD

**Issue by smoretti** [tadzik/panda#315]
Hi

When I try to use the option `--prefix=/path/to/modules` with panda, or with bootstrap.pl during panda installation, files are put at the place I use with prefix BUT they look like Perl5 files, they are not hashed and not visible as module by Perl6
E.g. `/usr/share/perl6/site/lib/URI/Escape.pm`

**Comment by niner** [comment]
/path/to/modules specifies a FileSystem repository path, same as with perl6 -I/path/to/modules.

Please try --prefix='inst#/path/to/modules' instead. This marks it as CompUnit::Repository::Installation

**Comment by smoretti** [comment]
Thanks, installation is done at the right place now.

Unfortunately I install packages as RPM, the installation process is done in a sandbox. panda installs the modules in the sandbox but cannot find them at the test time.
Ideally I would need an install path, and a install prefix path.

**Comment by niner** [comment]
I would very much like to talk to you about packaging issues on the #perl6-toolchain IRC channel. My rakudo work for the past few months has mostly been about enabling creation of RPM and other packages for modules. Precompilation makes this a very complicated topic. Just a couple of minutes ago I pushed a first version of a working packaging tool:
https://github.com/rakudo/rakudo/commit/2801780673f7f5a32f72e0a1d2b9a81e01f7f2e9

**Comment by smoretti** [comment]
I can see issues with precompilation. I work for an academic institute and we use RPM to keep track of dependencies for our field tools. We use Fedora RPM spec file for most OS based tools.

Perl6 is not very active (yet) in Fedora. Currently there are no Perl6 modules available there, nor panda, and rakudo is out-dated.
So I try to get few Perl6 modules (mainly those required by p6doc) to enable people here to play with Perl6

**Comment by smoretti** [comment]
Another issue for RPM is the panda/ directory that needs to be updated at each installation and may belong to different packages at the end (bad for RPM)

**Comment by niner** [comment]
That's why I don't recommend using panda for creating rpms. It was just not written for that job. The new tools/install-dist.pl that comes with rakudo (as of today) is much closer to what you need. The only part missing for now is support for Build.pm files, i.e. custom build steps. We still have to figure out a standardized way to handle those.

**Comment by smoretti** [comment]
Thanks, I will test install-dist.pl.

For now, I managed to build RPM for few modules I need with Panda. Just need to remove the precomp/ directory, as well as panda/projects.json and panda/state because they cannot be shared between RPMs and are complicated to update properly.
I use "panda  --nodeps" to install a single module per RPM (so need to check carefully and manually dependencies).
What is missing is maybe a way to install module from a local tarball or zip because some RPM build systems use a sandbox without network. Is it possible with install-dist.pl?

**Comment by niner** [comment]
Yes, install-dist.pl is a very basic tool. It installs an unpacked distribution. You use it like I wrote in the commit message:

VENDOR=$(perl6 -e 'CompUnit::RepositoryRegistry.repository-for-name("vendor").prefix.print')
perl6 install-dist.pl --for vendor --to /tmp/vendor /path/to/module

Then you can package the files in /tmp/vendor for installation into $VENDOR

This doesn't need any network access and the precomp files generated during this process can be used at runtime.

**Comment by smoretti** [comment]
OK, thanks
